src/Markup/Perspex.Markup/Data/Parsers/LiteralParser.cs

```diff
@@ -12,21 +12,33 @@ public static object Parse(Reader r)
         {
             if (char.IsDigit(r.Peek))
             {
-                StringBuilder result = new StringBuilder();
-
+                var result = new StringBuilder();
+                var foundDecimal = false;
                 while (!r.End)
                 {
                     if (char.IsDigit(r.Peek))
                     {
                         result.Append(r.Take());
                     }
+                    else if (!foundDecimal && r.Peek == '.')
+                    {
+                        result.Append(r.Take());
+                        foundDecimal = true;
+                    }
                     else
                     {
                         break;
                     }
                 }
 
-                return int.Parse(result.ToString(), CultureInfo.InvariantCulture);
+                if (!foundDecimal)
+                {
+                    return int.Parse(result.ToString(), CultureInfo.InvariantCulture); 
+                }
+                else
+                {
+                    return result.ToString(); // Leave as a string to support double, float, and decimal indicies
```

Comment:
Hmm, this strikes me as odd, that we'd return an integer if no decimal place was found but a string if it was. I understand why you did it like this, but maybe we should always return a string and convert later?